client/src/pages/profile.js

```diff
@@ -21,16 +22,53 @@ export default withStyles({
   }
 }, class Profile extends Component {
   state = {
-    name: 'Generic_CTFTeam'
+    dataLoading: true,
+    name: '',
+    division: '',
+    placement: '',
+    score: 0,
+    teamToken: '',
+    solves: []
   }
 
   componentDidMount () {
     document.title = 'Profile' + config.ctfTitle
-  }
 
-  render ({ classes }, { name }) {
-    const teamToken = localStorage.getItem('teamToken')
+    privateProfile()
+      .then(data => {
+        this.setState({
+          name: data.name,
+          division: data.division,
+          score: data.score.score,
+          teamToken: data.teamToken,
+          solves: data.returnedSolves
+        })
+        let placementStr = String(data.score.place)
+        if (data.score.place >= 11 && data.score.place <= 13) {
```

Comment:
This feels like it could go into a utility method

Reply:
This sounds right, I will move it.